instr/gen_decode.py

```diff
@@ -1,14 +1,137 @@
+import yaml
+import subprocess
 
-gen = open("decode.gen.cpp", "w")
-
-gen.write(
-    "#include <sim/instr.hpp>" "\n"
-    "\n"
-    "namespace sim {" "\n"
-    "namespace instr {" "\n"
-    "\n"
-    "Instr::Instr(InstrCode instr_code) {}" "\n"
-    "\n"
-    "}" "\n"
-    "}" "\n"
-)
+KNOWN_REGS = ["rd", "rs1", "rs2", "rm"]
+KNOWN_IMMS = [ "imm12", "imm20", "jimm20", "storeimm", "bimm", "shamt", "shamtw"]
+
+def GenerateGetBinValue(bit_section: dict) -> str:
+    write_buffer = ""
+    write_buffer += "bit::getBitField<InstrCode,"
+    write_buffer += f"{bit_section.get('msb')},"
+    write_buffer += f"{bit_section.get('lsb')}>(instr_code)"
+
+    shift = bit_section.get('to')
+    if shift != 0:
+        write_buffer += f" << {shift}"
+    return write_buffer
+
+
+def GenerateFieldAssigning(inst: dict, field_dict: dict) -> str:
+    write_buffer = ""
+    write_buffer += f"m_id = InstrId::{inst.get('mnemonic').upper().replace('.', '_', 3)};\n"
+    extend_from = 0 
+
+    for field in inst.get("fields"):
+        if field in KNOWN_REGS:
+            write_buffer += f"m_{field} = static_cast<uint8_t>("
+            write_buffer += GenerateGetBinValue(field_dict.get(field).get('location').get('bits')[0])
+            write_buffer += ");\n"
+        elif field in KNOWN_IMMS:
+            for bit_section in field_dict.get(field).get('location').get('bits'):
+                write_buffer += f"m_imm |= static_cast<uint32_t>("
+                extend_from = max(extend_from, bit_section.get('from'))
+                write_buffer += GenerateGetBinValue(bit_section)
+                write_buffer += ");\n"
+        
+        if field in KNOWN_IMMS and extend_from <= 30:
+            write_buffer += f"m_imm = bit::signExtend<InstrCode ,{extend_from + 1}>(m_imm);\n"
+    
+    return write_buffer
+
+
+def GenerateSwitchCaseMasks(yaml_dump: dict) -> str:
```

Comment:
Remove unused code, please

Reply:
This code might be used later for perf tests